### PR TITLE
[risk=no][RW-8705] BillingMigrationStatus: comments, removal, and deprecations

### PR DESCRIPTION
**Pull request by jmthibault79**
No functional changes expected.

Tested locally by making some workspace modifications.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/db/model/DbWorkspace.java [hunk 1]
```diff
@@ -32,13 +32,6 @@
 @Entity
 @Table(name = "workspace")
 public class DbWorkspace {
-  /**
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/model/DbWorkspace.java:35]: moved to end of file

File: api/src/main/java/org/pmiops/workbench/db/model/DbWorkspace.java [hunk 2]
```diff
@@ -74,13 +67,6 @@ public boolean equals(Object obj) {
     }
   }
 
-  // see https://precisionmedicineinitiative.atlassian.net/browse/RW-2705
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/model/DbWorkspace.java:77]: moved to end of file

File: api/src/main/java/org/pmiops/workbench/db/model/DbWorkspace.java [hunk 3]
```diff
@@ -387,15 +373,6 @@ public DbWorkspace setOtherPurposeDetails(String otherPurposeDetails) {
     return this;
   }
 
-  @Column(name = "billing_migration_status")
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/model/DbWorkspace.java:390]: moved to end

File: api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceDao.java [hunk 4]
```diff
@@ -103,18 +103,12 @@ Optional<DbWorkspace> findFirstByWorkspaceNamespaceAndActiveStatusOrderByLastMod
...
 
-  List<DbWorkspace> findAllByBillingMigrationStatus(Short billingMigrationStatus);
-
   DbWorkspace findDbWorkspaceByWorkspaceId(long workspaceId);
 
   Set<DbWorkspace> findAllByCreator(DbUser user);
 
   List<DbWorkspace> findAllByNeedsResearchPurposeReviewPrompt(short researchPurposeReviewed);
 
-  default List<DbWorkspace> findAllWithBillingMigrationNewStatus() {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceDao.java:114]: Removed in #6882

File: api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceDao.java [hunk 5]
```diff
@@ -103,18 +103,12 @@ Optional<DbWorkspace> findFirstByWorkspaceNamespaceAndActiveStatusOrderByLastMod
   Optional<DbWorkspace> findFirstByGoogleProjectAndActiveStatusOrderByLastModifiedTimeDesc(
       String googleProject, short activeStatus);
 
-  List<DbWorkspace> findAllByBillingMigrationStatus(Short billingMigrationStatus);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceDao.java:106]: only used by the below default method

File: api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceDao.java (hunk 4 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceDao.java:114]: and the final query relating to the field was removed in a later iteration of the same: #6922

File: api/src/test/java/org/pmiops/workbench/billing/FreeTierBillingServiceTest.java [hunk 6]
```diff
@@ -852,7 +852,6 @@ private DbUser createUser(String email) {
     return userDao.save(new DbUser().setUsername(email));
   }
 
-  // we only alert/record for BillingMigrationStatus.NEW workspaces
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/billing/FreeTierBillingServiceTest.java:855]: no longer used as a distinction as of #6922 - instead we use a deletion date cutoff, which enables us to do much less work

File: api/src/main/java/org/pmiops/workbench/db/model/DbWorkspace.java [hunk 7]
```diff
@@ -788,4 +765,41 @@ public int hashCode() {
         .append(googleProject)
         .toHashCode();
   }
+
+  /**
+   * Preserve the database migration field 'billing_migration_status' for historical purposes, to
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/model/DbWorkspace.java:770]: please let me know if you would like any text clarifications/improvements from here on down.

**Review by dmohs — APPROVED** [review]
Seems reasonably clear to me.

**Pull request merged**